src/main/scala/cognite/spark/v1/FileContentRelation.scala

```diff
@@ -0,0 +1,108 @@
+package cognite.spark.v1
+
+import cats.effect.{IO, Resource}
+import com.cognite.sdk.scala.v1.{FileDownloadExternalId, GenericClient}
+import fs2.Stream
+import org.apache.commons.io.FileUtils
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.sources.{BaseRelation, Filter, PrunedFilteredScan, TableScan}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Row, SQLContext, SparkSession}
+import org.apache.spark.{Partition, TaskContext}
+import sttp.client3.{HttpURLConnectionBackend, UriContext, basicRequest}
+
+import java.net.URL
+import scala.io.Source
+
+trait WithSizeLimit {
+  val sizeLimit: Long
+}
+
+class FileContentRelation(config: RelationConfig, fileExternalId: String)(
+    override val sqlContext: SQLContext)
+    extends BaseRelation
+    with TableScan
+    with PrunedFilteredScan
+    with Serializable
+    with WithSizeLimit {
+
+  override val sizeLimit: Long = 5 * FileUtils.ONE_GB
+
+  @transient lazy val client: GenericClient[IO] =
+    CdpConnector.clientFromConfig(config)
+
+  override def schema: StructType =
+    createDataFrame(sqlContext.sparkSession).schema
```

Comment:
will it read file in full? There's some streaming in how we fetch and pass around lines so might be better for mem consumption to do it twice, also should not be a problem for first iteration to read twice. But for bigger files future we might need to look into it (explicitly limit schema inference, or explicitly put schema as just jsons?)

Reply:
Yes, it will read it in full, as far as I understand it

Reply:
there is a setting to only read part of it if we want to (it's set as a percent)